### DISCUSSION TITLE
labels or label isn't an existing parameter for box_annotate()

### DISCUSSION THREAD

**Discussion by Jarradmorden** [roboflow/supervision#1634] [Q&A] [answered]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar bug report.

### Bug

  labels or label isn't an existing parameter for box_annotate(), what do we replace this with? is it needed I just removed it?

Notebook = https://github.com/roboflow/notebooks/blob/main/notebooks/how-to-auto-train-yolov8-model-with-autodistill.ipynb

box_annotator = sv.BoxAnnotator()
image = cv2.imread(f"C:/Users/jmorde02/DATA/2024-10-18_15-13-59\sample\left_3070_2024-10-18 15-17-30.481540.png")

mask_annotator = sv.MaskAnnotator()
results = base_model.predict(image)

annotated_image = mask_annotator.annotate(
	image.copy(), detections=results
)

images = []
for image_name in image_names:
    image  = dataset.images[image_name]
    annotations= dataset.annotations[image_name]
    labels = [
        dataset.classes[class_id]
        for class_id
        in annotations.class_id]
    annotates_image = mask_annotator.annotate(
        scene=image.copy(),
        detections=annotations)
    annotates_image = box_annotator.annotate(
        scene=annotates_image,
        detections=annotations
        labels=labels)
    images.append(annotates_image)

As you can see here the options

(method) def annotate(
    scene: ImageType@annotate,
    detections: Detections,
    custom_color_lookup: ndarray | None = None
) -> ImageType@annotate

### Environment

supervision==0.24.0
Windows 11
Python3.11
Cuda 11.8

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by LinasKo** [comment, marked as answer]
Hi @Jarradmorden 👋

The supervision version there is indeed very old. Here's how you'd use the annotators right now:

```python
mask_annotator = sv.MaskAnnotator()
box_annotator = sv.BoxAnnotator()
label_annotator = sv.LabelAnnotator()

images = []
for image_name, image, annotations in dataset:
    annotated_image = image.copy()
    annotated_image = mask_annotator.annotate(
        scene=annotated_image,
        detections=annotations)
    annotated_image = box_annotator.annotate(
        scene=annotated_image,
        detections=annotations)
    annotated_image = label_annotator.annotate(
        scene=annotated_image,
        detections=annotations,
    )
    images.append(annotated_image)
```

The `LabelAnnotator.annotate()` does accept a `labels` parameter, but you probably won't need it.

**Comment by onuralpszr** [comment]
Converting to Q&A since this is not a bug

**Reply by Jarradmorden** [reply]
Hi quick question because you're very helpful! Do you know if the ontology can be trained to do for example like "rear of the car" if I prompt it or does it have to be much more simpler and do a car, and if I then train it can I just include cars and it can work out what is the rear and segment it that way from the prompt, thanks!

**Reply by onuralpszr** [reply]
@Jarradmorden If you mean entire rear side  detect cars generally would be  fine, but you want to just detect for example  "car door" or something different you need to re-train a model, For prompt just "car" is simple for sure because you detect entire side so it could be easy and simple as well.

**Reply by Jarradmorden** [reply]
I really need just the rear side so should I do something like this 

"Rear end car": "rear end car" in the ontology  thanks very much!

I don't care about the entirety of the car

**Reply by onuralpszr** [reply]
Okay then sure do as you said 👍  but make sure test as well

**Reply by Jarradmorden** [reply]
I did test but because of low sample I couldn't tell if it was doing it at the time as it was only doing the car on front of me ! Hah thanks very much !

**Reply by onuralpszr** [reply]
@Jarradmorden  İf side note original question answer solve your problem please mark as answer 👍  That would be nice

**Reply by Jarradmorden** [reply]
Awesome will do thanks

**Reply by LinasKo** [reply]
To add more ideas into the mix, check out the dataset used for this one: https://github.com/roboflow/notebooks/blob/main/notebooks/train-yolov8-instance-segmentation-on-custom-dataset.ipynb
